doc/config/vhosts.d/vhosts.template

```diff
@@ -20,14 +20,15 @@ $HTTP["host"] == "download.example.com" {
   ##
   accesslog.filename          = log_root + "/" + server_name + "/access.log"
   server.errorlog             = log_root + "/" + server_name + "/error.log"
-}
 
-$SERVER["socket"] == "127.0.0.1:443" {
-  server.name = "localhost"
-  ssl.pemfile = "/etc/ssl/private/lighttpd-localhost.pem"
-  ssl.engine  = "enable"
+  $HTTP["scheme"] == "https" {
```

Comment:
Your suggested change appears to me to be to be redundant and possibly misleading.  If you have `ssl.engine = "enable"`, the scheme for requests should be `https`.  It is an error in lighttpd not to have a default certificate defined for a `$SERVER["socket"]` when `ssl.engine = "enable"` elsewhere in the config.  `ssl.engine = "enable"` should only appear in the global scope or in `$SERVER["socket"]`, and `$SERVER["socket"]` should not be nested inside other lighttpd conditions.

Related to my comment at the top of the review, the (original) example should probably be commented out, or else `server.modules += ( "mod_openssl" )` should be present.

On second thought, the original ssl example here was to illustrate that an alternative document root could be used, but I think this ssl section here should be removed entirely.

Reply:
Removed `ssl.engine = "enable"` from here and commented out rest of the example.

Reply:
I would like to remove the entire section.  You may do so as part of this patch, or I can do so separately.  If you are making changes, such as commenting the section out, please either justify why you are making changes here, or revert your changes here.

Reply:
I added this conditional way as example to easily have different certificates per vhost. I know there are other possible ways to implement this. I commented this out because as you stated that if turn on ssl.engine I must have a default certificate for the server.

Reply:
I wrote:
> On second thought, the original ssl example here was to illustrate that an alternative document root could be used, but I think this ssl section here should be removed entirely.

What was unclear?

You wrote:
> I added this conditional way as example to easily have different certificates per vhost. I know there are other possible ways to implement this.

The only relevant way to choose certificates based on vhost is `$HTTP["host"]`.  Every other way you can think of, including "scheme" is not based on vhost and therefore does not belong with the vhost config example.

Reply:
Sorry, my bad. I have removed that part completely.